scripts/workflows/provision-darwin.sh

```diff
@@ -34,3 +34,5 @@ echo "BATSLIB=${BATS_SUPPORT}" >> "$GITHUB_ENV"
 
 # Exit temporary directory.
 popd
+
+cargo install ic-cdk-optimizer
```

Comment:
This takes a really long time.  Can we install this only for the rust tests?

Reply:
I intended to have the `ic-cdk-optimizer` binary be cached. Can we?

Reply:
Sure, we can give that a shot.  We just need to be sure we aren't adding 12 minutes to the build time of each matrix job.  A place to start: right now, the `test` job, which runs this provisioning and the e2e tests themselves, does not have a cargo cache step.

https://github.com/dfinity/sdk/blob/master/.github/workflows/e2e.yml#L116

It also hasn't configured rust to any particular version, as is done in the `build` job here: https://github.com/dfinity/sdk/blob/master/.github/workflows/e2e.yml#L30

Reply:
I'm moving this `cargo install ic-cdk-optimizer` mainly for overcoming the 1200s timeout on macOS.
https://github.com/dfinity/sdk/runs/4462587938?check_suite_focus=true
May I increase this limit?